### PR TITLE
Update: Remove bard-request models and load report

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
Relatively big PR that removes all v1 bard-request models, and adds basic functionality so that loading a report will work. Our next focus is reworking visualizations/etc places that follow fili assumptions of having a `dateTime` which specifies the time grain as well as the interval

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/dimension-selector.js [hunk 1]
```diff
@@ -62,44 +63,29 @@ export default class DimensionSelector extends Component {
    */
   classNames = ['checkbox-selector', 'checkbox-selector--dimension'];
 
+  @service bardMetadata;
+
+  @computed('request.{dataSource,table}')
+  get currentTable() {
+    const { dataSource, table } = this.request;
+    return this.bardMetadata.getById('table', table, dataSource);
```
Comment (jkusa) [inline packages/reports/addon/components/dimension-selector.js:71]: //TODO use model, the model doesn't have the metadata

File: packages/core/addon/models/table.js [hunk 2]
```diff
@@ -135,15 +135,17 @@ function hasAllColumns(request, columns) {
         }
       }),
     dimensions = [].concat(
-      ...get(request, 'dimensions').map(dimension => {
-        let name = dimension.dimension.id,
-          defaultFields = getDefaultDimensionFields(dimension);
-        return !defaultFields.length ? name : defaultFields.map(field => canonicalizeDimension({ id: name, field }));
-      })
+      ...request.columns
```
Comment (jkusa) [inline packages/core/addon/models/table.js:138]: Maybe we should have a `dimensionColumns` cp on the request model

File: packages/core/addon/chart-builders/metric.js [hunk 3]
```diff
@@ -51,9 +51,10 @@ export default EmberObject.extend({
     // Support different `dateTime` formats by mapping them to a standard
     const buildDateKey = dateTime => moment(dateTime).format(DateUtils.API_DATE_FORMAT_STRING);
 
-    let metrics = config.metrics,
-      grain = request.logicalTable?.timeGrain,
-      requestInterval = Interval.parseFromStrings(request.intervals?.[0]?.start, request.intervals?.[0]?.end);
+    const metrics = config.metrics;
+    const grain = request.columns.find(f => f.type === 'time-dimension' && f.field === 'dateTime').parameters.grain;
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/core/addon/chart-builders/metric.js:55]: we have https://github.com/yahoo/navi/blob/request-v2/packages/core/addon/models/bard-request-v2/request.js#L134

File: packages/reports/addon/components/dimension-selector.js [hunk 4]
```diff
@@ -108,59 +94,48 @@ export default class DimensionSelector extends Component {
...
   }
 
   /*
    * @property {Array} selectedDimensions - dimensions in the request
    */
-  @mapBy('request.dimensions', 'dimension')
-  selectedDimensions;
+  @computed('request.columns.[]')
+  get selectedDimensions() {
+    return this.request.columns.filter(IS_DIMENSION);
```
Comment (jkusa) [inline packages/reports/addon/components/dimension-selector.js:112]:
```js
this.request.dimensionColumns
```

File: packages/reports/addon/components/dimension-selector.js [hunk 5]
```diff
@@ -108,59 +94,48 @@ export default class DimensionSelector extends Component {
...
   get selectedTimeGrain() {
-    const {
-      allTimeGrains,
-      request: {
-        logicalTable: { timeGrain }
-      }
-    } = this;
-
-    return allTimeGrains.find(grain => grain.id === timeGrain);
+    const { currentTable, request } = this;
```
Comment (jkusa) [inline packages/reports/addon/components/dimension-selector.js:128]: do we still need this with request preview enabled?

File: packages/reports/addon/components/filter-values/base-interval-component.js [hunk 6]
```diff
@@ -37,15 +38,18 @@ export default class BaseIntervalComponent extends Component {
...
 
   /**
    * @property {String} lowestDateTimePeriod - the lowest supported time grain of the table
    */
   @computed('request.logicalTable.table.timeGrainIds.[]')
   get lowestDateTimePeriod() {
-    const { timeGrainIds } = this.request.logicalTable.table;
-    const sorted = [...timeGrainIds].sort((lId, rId) => {
+    const { table, dataSource } = this.request;
+    const tableMetadata = this.bardMetadata.getById('table', table, dataSource);
```
Comment (jkusa) [inline packages/reports/addon/components/filter-values/base-interval-component.js:51]: can we use the model?

File: packages/reports/addon/components/metric-selector.js [hunk 7]
```diff
@@ -62,10 +70,11 @@ class MetricSelectorComponent extends Component {
    * @property {Object} metricsFiltered - metric -> boolean mapping denoting presence of metric
    *                                         in request havings
    */
-  @computed('request.having.[]')
+  @computed('request.filters.[]')
   get metricsFiltered() {
-    return A(get(this, 'request.having'))
-      .mapBy('metric.metric.id')
+    return this.request.filters
+      .filter(f => f.type === 'metric')
```
Comment (jkusa) [inline packages/reports/addon/components/metric-selector.js:76]: `metricFilters` could be a cp on the request

File: packages/reports/addon/components/navi-list-selector.js [hunk 8]
```diff
@@ -47,41 +42,22 @@ export default Component.extend({
...
-
-    //set items to selected or all items based on showSelected
-    if (get(this, 'showSelected')) {
-      items = get(this, 'selected');
-    } else {
-      items = get(this, 'items');
-    }
+  filteredItems: computed('items', 'query', 'searchField', 'selected', function() {
+    const query = get(this, 'query');
+    const items = get(this, 'items');
```
Comment (jkusa) [inline packages/reports/addon/components/navi-list-selector.js:54]: need the `get`?

File: packages/reports/addon/components/report-builder.js [hunk 9]
```diff
@@ -42,6 +40,12 @@ export default class ReportBuilderComponent extends Component {
     return this.metadataService.all('table').sortBy('name');
   }
 
+  @computed('request.{dataSource,table}')
+  get currentTable() {
+    const { dataSource, table } = this.request;
+    return this.metadataService.getById('table', table, dataSource);
```
Comment (jkusa) [inline packages/reports/addon/components/report-builder.js:46]: same

File: packages/reports/addon/components/report-builder.js [hunk 10]
```diff
@@ -79,7 +83,7 @@ export default class ReportBuilderComponent extends Component {
    */
   @action
   onToggleMetricFilter(metric) {
-    this._expandFilters(() => (this.request.having || []).find(having => having.metric.metric.name === metric.name));
+    this._expandFilters(() => (this.request.filters || []).find(f => f.type === 'metric' && f.field === metric.name));
```
Comment (jkusa) [inline packages/reports/addon/components/report-builder.js:86]: cp's?

File: packages/reports/addon/components/dimension-selector.js (hunk 5 above)
Comment (kevinhinterlong) [reply packages/reports/addon/components/dimension-selector.js:128]: https://github.com/yahoo/navi/blob/178c82dc1b9b1cd353f055376b37c6cbcda3615f/packages/reports/addon/templates/components/dimension-selector.hbs#L18
It's used in this block, so once we better integrate timeDimension we could fix it

File: packages/core/addon/models/table.js (hunk 2 above)
Comment (kevinhinterlong) [reply packages/core/addon/models/table.js:139]: added a todo for this one

File: packages/core/addon/models/bard-request-v2/request.js [hunk 11]
```diff
@@ -119,6 +121,43 @@ export default class Request extends Fragment.extend(Validations) {
...
+  get metricColumns() {
+    return this.columns.filter(column => column.type === 'metric');
+  }
+
+  /**
+   * @property {ColumnFragment[]} dimensionColumns - The dimension columns
+   */
+  @computed('columns.[]')
+  get dimensionColumns() {
+    return this.columns.filter(column => column.type === 'dimension');
```
Comment (jkusa) [inline packages/core/addon/models/bard-request-v2/request.js:142]: do we want this to include `time-dimension`s

File: packages/reports/addon/templates/components/metric-selector.hbs [hunk 12]
```diff
@@ -3,7 +3,7 @@
   <NaviListSelector
     @items={{this.allMetrics}}
     @searchField="name"
-    @title="Metrics"
+    @title="Metrics"q
```
Comment (jkusa) [inline packages/reports/addon/templates/components/metric-selector.hbs:6]: ?

**Pull request merged**